src/magic_folder/config.py

```diff
@@ -0,0 +1,419 @@
+            cursor.execute("SELECT tahoe_client_url FROM config")
+            return cursor.fetchone()[0]
+
+    @tahoe_client_url.setter
+    def tahoe_client_url(self, url):
+        with self.database:
+            cursor = self.database.cursor()
+            cursor.execute("UPDATE config SET tahoe_client_url=?", (url, ))
+
+    def get_magic_folder(self, name):
+        """
+        Find the config for an existing Magic Folder.
+
+        :param unicode name: the unique name of the magic-folder to find
+
+        :returns: a MagicFolderConfig instance
+
+        :raises: ValueError if there is no such Magic Folder
+        """
+        with self.database:
+            cursor = self.database.cursor()
+            cursor.execute("SELECT name, location FROM magic_folders WHERE name=?", (name, ))
+            data = cursor.fetchone()
+            if data is None:
+                raise ValueError(
+                    "No Magic Folder named '{}'".format(name)
+                )
+            name, location = data
+            connection = sqlite3.connect(location)
+            config = MagicFolderConfig(
+                name=name,
+                database=connection,
+            )
+            return config
+
+    def create_magic_folder(self, name, magic_path, state_path, author,
+                            collective_dircap, upload_dircap, poll_interval):
+        """
+        Add a new Magic Folder configuration.
+
+        :param unicode name: a unique name for this magic-folder
+
+        :param FilePath magic_path: the synchronized directory which
+            must already exist.
+
+        :param FilePath state_path: the configuration and state
+            directory (which should not already exist)
+
+        :param LocalAuthor author: the signer of snapshots created in
+            this folder
+
+        :param unicode collective_dircap: the read-capability of the
+            directory defining the magic-folder.
+
+        :param unicode upload_dircap: the write-capability of the
+            directory we upload data into.
+
+        :param FilePath magic_directory: local path to the folder we
+            synchronize for this magic-folder.
+
+        :returns: a MagicFolderConfig instance
+        """
+        with self.database:
+            cursor = self.database.cursor()
+            cursor.execute("SELECT name FROM magic_folders WHERE name=?", (name, ))
+            if len(cursor.fetchall()):
+                raise ValueError(
+                    "Already have a magic-folder named '{}'".format(name)
+                )
+        if not magic_path.exists():
+            raise ValueError(
+                "'{}' does not exist".format(magic_path.path)
+            )
+        if state_path.exists():
+            raise ValueError(
+                "'{}' already exists".format(state_path.path)
+            )
+
+        mkdir(state_path.path)
+        stash_path = state_path.child("stash")
+        mkdir(stash_path.path)
```

Comment:
It would be nice to (eventually) make this function atomic to simplify error recovery.  If we manage to make the state path and/or the stash path but then fail to populate the database schema, the magic-folder has not been created but you cannot try again with `create_magic_folder` without doing some manual cleanup of what should be internal state.